### ISSUE TITLE
pylint should support recognizing type stubs with @overload

### ISSUE THREAD

**Issue by jnsnow** [PyCQA/pylint#3421]
When using MyPy and pylint together, you may run into cases where it is necessary to use functionality like typing.overload to decorate a function to give it more precise type information.

Consider the following snippet:
```python
T = TypeVar('T')

@overload
def localize(v: str) -> str: ...

@overload
def localize(v: T) -> T: ...

def localize(v: T) -> Union[str, T]:
    if isinstance(v, str):
        return loctable.detokenize(I18NString(v), default='')
    return v
```

This strategy is necessary because mypy cannot dynamically constrain the return type to understand which type of the Union is being returned. It's legitimate to define this overloading directly inside of the module instead of a type stub.

In this case, however, pylint is confused about the nature of the stub:
```
export.py:59:37: C0321: More than one statement on a single line (multiple-statements)
```

I would like to request that pylint is able to suppress this warning for type stub definitions in cases where it's obvious that it's a stub. `@overload` is a really good candidate: these will always be type stubs.

See also: https://docs.python.org/3/library/typing.html#typing.overload

This is related to #2873 but it's not identical: I am not linting a stub file, I am linting code with stubs in them. In this case, the only change needed (I think) is an exception for a multi-line statement when:

1. A preceding statement is a type stub decorator
2. There are precisely two statements on a line
3. The first statement is a `def`
4. The second statement is the ellipsis literal `...`

**Comment by PCManticore** [comment]
This is already fixed by https://github.com/PyCQA/pylint/commit/b63a385a5d8f287aaa2e209d199e4997baf962ea which will be part of pylint 2.5 when released.
